src/factory/service.ts

```
@@ -0,0 +1,77 @@
+import contentDisposition from 'content-disposition';
+import { Response, NextFunction } from 'express';
+import request from 'request';
+
+import { logger } from '../utils/Logger';
+import { bufferRequestBody } from './util';
+
+export enum DownloadType {
+  PDF = 'pdf',
+  XLSX = 'xlsx',
+}
+
+export enum XLSXType {
+  PROPOSAL = 'proposal',
+  SEP = 'sep',
+}
+
+export enum PDFType {
+  PROPOSAL = 'proposal',
+  SAMPLE = 'sample',
+}
+
+export type MetaBase = { collectionFilename: string; singleFilename: string };
+export type XLSXMetaBase = MetaBase & { columns: string[] };
+
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+const ENDPOINT = process.env.USER_OFFICE_FACTORY_ENDPOINT!;
```

I think we should either thrown an error or handle the case when the endpoint is not provided by returning an error to the client then the request cant be handled because of the missing endpoint

---

Well, yeah, if the endpoint doesn't exist then the logs will contain the error, but it's impossible to inform the FE what was the error unless we create an endpoint which is used for checking if the endpoint is defined. 

This should be defined every time, even if it doesn't point anywhere, maybe it would be better to just crash the backend if it doesn't exist.

---

Maybe we should go with "Fail Fast, Fail Often"